src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -97,25 +122,28 @@ async def pylint_lint(
             output_filename="requirements.pex",
             requirements=PexRequirements.create_from_requirement_fields(
                 tgt[PythonRequirementsField]
-                for tgt in targets
+                for tgt in targets_with_dependencies
                 if tgt.has_field(PythonRequirementsField)
             ),
             interpreter_constraints=interpreter_constraints,
         )
     )
+    # TODO(John Sirois): Support shading python binaries:
+    #   https://github.com/pantsbuild/pants/issues/9206
+    # Right now any Pylint transitive requirements will shadow corresponding user
+    # requirements, which could lead to problems.
+    pylint_runner_pex_args = ["--pex-path", ":".join(["pylint.pex", "requirements.pex"])]
+    if pylint.source_plugins:
+        # NB: Pylint requires explicitly loading source plugins through PYTHONPATH. To do this, we
+        # set `PYTHONPATH: ./`. But, PEX normally ignores PYTHONPATH for hermiticity, so we must
```

Comment:
This comment went out of sync with the other location. Consider combining the explanation in one location and then referring to it from both consumers.